pydantic_extra_types/coordinate.py

```diff
@@ -0,0 +1,202 @@
+from pydantic._internal import _repr
+from pydantic.json_schema import JsonSchemaValue
+from pydantic_core import CoreSchema, PydanticCustomError, core_schema
+
+LatitudeType = Union[str, int, float]
+
+
+class Latitude(float):
+    ge: ClassVar[float] = -90.00
+    le: ClassVar[float] = 90.00
+
+    def __init__(self, value: LatitudeType):
+        self.validate_lat(value)
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: Type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        return core_schema.general_after_validator_function(
+            cls._validate, core_schema.float_schema(ge=cls.ge, le=cls.le)
+        )
+
+    @classmethod
+    def _validate(cls, __input_value: Any, _: core_schema.ValidationInfo) -> 'Latitude':
+        return cls(__input_value)
+
+    @classmethod
+    def validate_lat(cls, value: LatitudeType) -> None:
+        """
+        Validate the latitude value.
+
+        Raises:
+            PydanticCustomError: If the latitude value is not within the valid range.
+        """
+        try:
+            if float(value) > cls.le or float(value) < cls.ge:
+                raise PydanticCustomError('latitude_error', 'Latitude must be between -90 and 90')
+        except ValueError:
+            raise PydanticCustomError(
+                'latitude_error',
+                'value is not a valid Latitude: value must be a str, int or float',
+            )
+
+
+LongitudeType = Union[str, int, float]
+
+
+class Longitude(float):
+    ge: ClassVar[float] = -180.00
+    le: ClassVar[float] = 180.00
+
+    def __init__(self, value: LongitudeType):
+        self.validate_long(value)
+
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: Type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        return core_schema.general_after_validator_function(
+            cls._validate, core_schema.float_schema(ge=cls.ge, le=cls.le)
+        )
+
+    @classmethod
+    def _validate(cls, __input_value: Any, _: core_schema.ValidationInfo) -> 'Longitude':
+        return cls(__input_value)
+
+    @classmethod
+    def validate_long(cls, value: Any) -> None:
+        """
+        Validate the longitude value.
+
+        Raises:
+            PydanticCustomError: If the longitude value is not within the valid range.
+        """
+        try:
+            if float(value) > cls.le or float(value) < cls.ge:
+                raise PydanticCustomError('longitude_error', 'Longitude must be between -180 and 180')
+        except ValueError:
+            raise PydanticCustomError(
+                'longitude_error',
+                'value is not a valid Longitude: value must be a str, int or float',
+            )
+
+
+CoordinateTuple = Tuple[Latitude, Longitude]
+CoordinateType = Union[CoordinateTuple, str, 'Coordinate']
+
+
+class Coordinate(_repr.Representation):
+    __slots__ = '_latitude', '_longitude'
+
+    def __init__(self, value: CoordinateType) -> None:
```

Comment:
Shouldn't this be a plain validator used in `__get_pydantic_core_schema__` instead of `__init__`?

Reply:
I've tried to base on the current implementation of other types. 
Do you have an example of how to validate only with the function `__get_pydantic_core_schema__`?

Reply:
Maybe, something like this https://github.com/pydantic/pydantic/blob/d32790445a596e45266a4636f8ec9f8e5434e83a/pydantic/types.py#L927

Reply:
it seems less obvious than in the example, as this is a `_repr.Representation`. Can we keep the __init__ function?